src/main/java/org/jvnet/hudson/test/JavaNetReverseProxy.java

```diff
@@ -61,9 +61,11 @@ protected void service(HttpServletRequest req, HttpServletResponse resp) throws
         String d = Util.getDigestOf(path);
 
         File cache = new File(cacheFolder, d);
-        if(!cache.exists()) {
-            URL url = new URL("https://updates.jenkins.io/" + path);
-            FileUtils.copyURLToFile(url,cache);
+        synchronized(this) {
+            if (!cache.exists()) {
+                URL url = new URL("https://updates.jenkins.io/" + path);
+                FileUtils.copyURLToFile(url,cache);
+            }
         }
```

Comment:
Unrelated, but @jtnord noticed the race condition here when discussing these changes.